tests/TileDB.CSharp.Test/ArrayTest.cs

```diff
@@ -123,11 +124,54 @@ public void TestSparseArray()
             Assert.IsNotNull(array_schema_loaded);
         }
 
+        [TestMethod]
+        public void TestConsolidateFragments()
+        {
+            const uint FragmentCount = 10;
+
+            var context = Context.GetDefault();
+
+            using var uri = new TemporaryDirectory("array_consolidate_fragments");
+
+            using (var schema = BuildDenseArraySchema(context))
+            using (var array = new Array(context, uri))
+            {
+                array.Create(schema);
+            }
+
+            int[] a1Data = Enumerable.Range(1, 10).Select(x => x * x).ToArray();
+            byte[] a2Data = "aabbcccdeffggghhhhijj"u8.ToArray();
+            ulong[] a2Offsets = { 0, 2, 4, 7, 8, 9, 11, 14, 17, 18 };
+            for (uint i = 0; i < FragmentCount; i++)
+            {
+                using var array = new Array(context, uri);
+                array.Open(QueryType.Write);
+
+                using var query = new Query(context, array, QueryType.Write);
+                query.SetDataBuffer("a1", a1Data);
+                query.SetDataBuffer("a2", a2Data);
+                query.SetOffsetsBuffer("a2", a2Offsets);
+
+                query.Submit();
+            }
+
+            using var fragmentInfo = new FragmentInfo(context, uri);
+            fragmentInfo.Load();
+
+            Assert.AreEqual(FragmentCount, fragmentInfo.FragmentCount);
+
+            string[] fragmentUris = Enumerable.Range(0, (int)FragmentCount).Select(x => fragmentInfo.GetFragmentName((uint)x)).ToArray();
+
+            Array.ConsolidateFragments(context, uri, fragmentUris);
+
+            fragmentInfo.Load();
+
+            Assert.AreEqual(1u, fragmentInfo.FragmentCount);
```

Comment:
Please add check of `to_vacuum_num == 10` here

Reply:
Done.